### ISSUE TITLE
Could not find the HAAR face detector location and OpenCV 'getData' error

### ISSUE THREAD

**Issue by Dongagent** [TadasBaltrusaitis/OpenFace#912]
**Describe the bug**
## Bug 1
**Could not find the HAAR face detector location**

## Bug 2
(tracking result:26321): Gtk-WARNING **: 02:57:34.702: cannot open display:
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.1.0) /home/src/opencv-4.1.0/modules/core/src/system.cpp:1533: error: (-215:Assertion failed) key_ != -1 && "Can't fetch data from terminated TLS container." in function 'getData'

**To Reproduce**

I follow the instructions on this site ([Link](https://github.com/TadasBaltrusaitis/OpenFace/wiki/Unix-Installation)) step by step and installed the packages successfully. In the meantime, I downloaded the model by this ([Link](https://github.com/TadasBaltrusaitis/OpenFace/wiki/Model-download))

Then I use the following code in the `/path/to/Openface/build`
`./bin/FaceLandmarkVid -f "../samples/changeLighting.wmv" -f "../samples/2015-10-15-15-14.avi"`

**log**
root@5798f27bdd7d:/home/src/OpenFace/build# ./bin/FaceLandmarkVid -f "../samples/changeLighting.wmv" -f "../samples/2015-10-15-15-14.avi"
Could not find the HAAR face detector location
Reading the landmark detector/tracker from: ./bin/model/main_ceclm_general.txt
Reading the landmark detector module from: ./bin/model/cen_general.txt
Reading the PDM module from: ./bin/model/pdms/In-the-wild_aligned_PDM_68.txt....Done
Reading the Triangulations module from: ./bin/model/tris_68.txt....Done
Reading the intensity CEN patch experts from: ./bin/model/patch_experts/cen_patches_0.25_of.dat....Done
Reading the intensity CEN patch experts from: ./bin/model/patch_experts/cen_patches_0.35_of.dat....Done
Reading the intensity CEN patch experts from: ./bin/model/patch_experts/cen_patches_0.50_of.dat....Done
Reading the intensity CEN patch experts from: ./bin/model/patch_experts/cen_patches_1.00_of.dat....Done
Reading part based module....left_eye_28
Reading the landmark detector/tracker from: ./bin/model/model_eye/main_clnf_synth_left.txt
Reading the landmark detector module from: ./bin/model/model_eye/clnf_left_synth.txt
Reading the PDM module from: ./bin/model/model_eye/pdms/pdm_28_l_eye_3D_closed.txt....Done
Reading the intensity CCNF patch experts from: ./bin/model/model_eye/patch_experts/left_ccnf_patches_1.00_synth_lid_.txt....Done
Reading the intensity CCNF patch experts from: ./bin/model/model_eye/patch_experts/left_ccnf_patches_1.50_synth_lid_.txt....Done
Could not find the HAAR face detector location
Done
Reading part based module....right_eye_28
Reading the landmark detector/tracker from: ./bin/model/model_eye/main_clnf_synth_right.txt
Reading the landmark detector module from: ./bin/model/model_eye/clnf_right_synth.txt
Reading the PDM module from: ./bin/model/model_eye/pdms/pdm_28_eye_3D_closed.txt....Done
Reading the intensity CCNF patch experts from: ./bin/model/model_eye/patch_experts/ccnf_patches_1.00_synth_lid_.txt....Done
Reading the intensity CCNF patch experts from: ./bin/model/model_eye/patch_experts/ccnf_patches_1.50_synth_lid_.txt....Done
Could not find the HAAR face detector location
Done
Reading the landmark validation module....Done
Attempting to read from file: ../samples/changeLighting.wmv
Device or file opened
Starting tracking
Reading the MTCNN face detector from: ./bin/model/mtcnn_detector/MTCNN_detector.txt
Reading the PNet module from: ./bin/model/mtcnn_detector/PNet.dat
Reading the RNet module from: ./bin/model/mtcnn_detector/RNet.dat
Reading the ONet module from: ./bin/model/mtcnn_detector/ONet.dat

(tracking result:27657): Gtk-WARNING **: 04:26:30.883: cannot open display:

**Desktop (please complete the following information):**
 - OS: Ubuntu 
 - Version 18.04

**Additional context**
Could you please tell me how to solve this? I have seen most of the issues.

**Comment by TadasBaltrusaitis** [comment]
Are you doing this from a machine with a display connected? This seems to be an issue with ability to open the display rather than the HAAR face detector (which is not used by default).

**Comment by deinhofer** [comment]
Hi,
I just had the same issue. it seems that under Linux the classifier folder is missing.
If you download the binary release ```.zip``` file for windows, simply copy the classifier folder to your Linux ```build/bin``` folder.

But as @TadasBaltrusaitis says the HAAR problem does not seem to be the main problem. You need a display to run the GUI sw.

**Comment by Dongagent** [comment]
@TadasBaltrusaitis @deinhofer Thank you very much for your information. Yes, I'm using the Linux server **without** a display connected. So I was wondering whether there’s an option to close the GUI function, then the HAAR problem would not occur. I hope this error had no effect on the final result of the video processing.

**Comment by TadasBaltrusaitis** [comment]
FeatureExtraction executable does not use a GUI by default, so you should be able to run that on a server without a display.

**Comment by ucalyptus2** [comment]
@TadasBaltrusaitis can a flag be created to do GUI=OFF for FaceLandmarkVid?

**Comment by muximuxi** [comment]
same error， could you please tell how to solve the problem?

**Comment by brmarkus** [comment]
@muximuxi can you provide more details, please?
Which environment, what have you done, how have you tried it? What does the error look like (logs, console)?
Same - regarding display, regarding detector location (using the downloaded ZIP and copied it?)?

**Comment by muximuxi** [comment]
> 
I use Ubuntu 16.04.4, and also downloaded ZIP and copied it. The error reported is：
Could not find the HAAR face detector location
Done
Reading the landmark validation module....Done
Attempting to read from file: ../samples/2015-10-15-15-14.avi
Device or file opened
Starting tracking
Reading the MTCNN face detector from: ./bin/model/mtcnn_detector/MTCNN_detector.txt
Reading the PNet module from: ./bin/model/mtcnn_detector/PNet.dat
Reading the RNet module from: ./bin/model/mtcnn_detector/RNet.dat
Reading the ONet module from: ./bin/model/mtcnn_detector/ONet.dat
(tracking result:26499): Gtk-WARNING **: cannot open display:
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.1.0) /home/opencv-4.1.0/modules/core/src/system.cpp:1533: error: (-215:Assertion failed) key_ != -1 && "Can't fetch data from terminated TLS container." in function 'getData'

**Comment by brmarkus** [comment]
What ZIP from where (which version) have you downloaded - and what have you copied to which location?
From which location have you called which application using which command line parameters?

You also have no display connected (`Gtk-WARNING **: cannot open display`)? Or are you connected remotely to the test machine withtout setting the `DISPLAY` env variable?

**Comment by muximuxi** [comment]
Yes, I'm connected remotely machine ,and use ./bin/FaceLandmarkVid -f "../samples/2015-10-15-15-14.avi" 
how to set DISPLAY env variable? use -DISPLAY False ?
And I followed the installation instructions exactly

**Comment by brmarkus** [comment]
> Hi, I just had the same issue. it seems that under Linux the classifier folder is missing. If you download the binary release `.zip` file for windows, simply copy the classifier folder to your Linux `build/bin` folder.
> 
> But as @TadasBaltrusaitis says the HAAR problem does not seem to be the main problem. You need a display to run the GUI sw.

@muximuxi could this be your problem as well?

**Comment by brmarkus** [comment]
> FeatureExtraction executable does not use a GUI by default, so you should be able to run that on a server without a display.

@muximuxi Could you also use `FeatureExtraction` instead of `FaceLandmarkVid`?

Seems like nobody has added a flag to disable the GUI - in case of no display, no monitor or no display-forwarding.
